core/src/main/kotlin/org/evomaster/core/output/TestSuiteOrganizer.kt

```diff
@@ -13,22 +20,131 @@ import org.evomaster.core.search.Solution
  * name each test will have
  */
 class TestSuiteOrganizer {
-
     companion object {
-        fun sortTests(solution: Solution<*>): List<TestCase> {
+        val sortingHelper = SortingHelper()
+        val namingHelper = NamingHelper()
+
+
+        fun sortTests(solution: Solution<*>, customNaming: Boolean = false): List<TestCase> {
 
             //TODO here in the future we will have something bit smarter
-            return naiveSorting(solution)
+            return sortingHelper.sort(solution, namingHelper, customNaming)
+        }
+    }
+}
+
+class NamingHelper {
+    private fun criterion1_500 (individual: EvaluatedIndividual<*>): String{
+        var found500 = false
+        individual.results.forEach {
+            if((it as RestCallResult).getStatusCode() == 500) found500 = true
+        }
+        val suggestedName = when (found500) {
+            true ->  "_with500"
+            false ->  ""
+            else -> throw IllegalStateException("This should not happen")
+        }
+        return suggestedName
+    }
+
+    private fun criterion2_hasPost (individual: EvaluatedIndividual<*>): String{
+        var foundPost = false
+        individual.individual.seeActions().forEach {
```

Comment:
looks inefficient. better to use ".any{predicate}" instead of "forEach" here (apply this also on other similar code in this file)